browser/src/react-redux/App/SignupPage.jsx

```diff
@@ -0,0 +1,159 @@
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+import setUser from './setUser.action'
+import axios from 'axios';
+
+const initialState = {
+  confirmPassword: '',
+  email: '',
+  password: '',
+  emailError: '',
+  generalError: '',
+  passwordError: ''
+};
+
+const mapDispatchToProps = dispatch => {
+  return {
+    setUser: user => {
+      dispatch(setUser(user))
+    }
+  }
+}
+
+const mapStateToProps = state => {
+  return { user: state.user }
+}
+
+class SignupPage extends Component {
+  static propTypes = {
+    history: PropTypes.object.isRequired,
+    setUser: PropTypes.func.isRequired,
+    user: PropTypes.shape({
+      email: PropTypes.string
+    })
+  };
+
+  state = initialState
+
+  handleChange = (event) => {
+    this.setState({
+      [event.target.name]: event.target.value
+    });
+  }
+
+  getConfirmPasswordError = () => {
+    return (this.state.confirmPassword && this.state.password !== this.state.confirmPassword)
+      ? 'Passwords do not match.'
+      : null;
+  }
+
+  enableButton = () => {
```

Comment:
Because this is specific to the submit button, let's be clear about that and call this method `enableSubmitButton`.